umodbus/functions.py

```diff
@@ -672,12 +706,18 @@ def create_from_request_pdu(cls, pdu):
 
         fmt = '>' + ('B' * byte_count)
         values = struct.unpack(fmt, pdu[6:])
-        values = [integer_to_binary_list(v) for v in values]
 
-        # Flatten list
-        values = [n for value in values for n in value]
+        res = list()
 
-        return cls(starting_address, quantity, byte_count, values)
+        for i, value in enumerate(values):
```

Comment:
I don't understand what is magic about it. It's an iteration of a list of values like: `[6, 144, 255]`.